ui/text_input_modal.go

```diff
@@ -0,0 +1,98 @@
+package ui
+
+import (
+	"fmt"
+
+	"gioui.org/layout"
+	"gioui.org/widget"
+	"github.com/planetdecred/godcr/ui/decredmaterial"
+)
+
+const ModalTextInput = "text_input_modal"
+
+type textInputModal struct {
+	*infoModal
+
+	isLoading bool
+
+	textInput decredmaterial.Editor
+	callback  func(string, *textInputModal) bool
```

Comment:
a comment should probably be added that the callback here and in other modals where they appear would be optional on some modals and the main aim of the modal is to have it handle their intended functions within the modal.